src/components/SpruceForm/FieldTemplates.tsx

```diff
@@ -291,11 +299,18 @@ export const ArrayFieldTemplate: React.VFC<ArrayFieldTemplateProps> = ({
             />
           ))
         )}
+        {!readonly && canAdd && addToEnd && (
```

Comment:
We can extract these repeated conditionals into a common variable to make this more readable.

```ts
L257: const buttonAtEnd = !readOnly && canAdd && addToEnd;
...
L280: {!buttonAtEnd && (<AddButt.... AddButtonContainer>)
...
L302: {buttonAtEnd && (<AddButt.... AddButtonContainer>)
```